core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1593,17 +1600,15 @@ public TableMetadata build() {
           discardChanges ? ImmutableList.of() : ImmutableList.copyOf(changes));
     }
 
+    // TODO: remove newLastColumnId in favor of schema.highestFieldId() once
```

Comment:
nit: maybe move this comment before line 1608 so it's closer to the line we want to remove